lib/engine/step/tracker.rb

```diff
@@ -91,9 +91,20 @@ def lay_tile(action, extra_cost: 0, entity: nil, spender: nil)
         end
 
         entity.abilities(:teleport) do |ability, _|
-          ability.use! if ability.hexes.include?(hex.id) && ability.tiles.include?(tile.name)
+          next if !ability.hexes.include?(hex.id) || !ability.tiles.include?(tile.name)
+
+          teleport = true
+          free = true if ability.free_tile_lay
+          if ability.cost
+            spender.spend(ability.cost, @game.bank) if ability.cost.positive?
+            @log << "#{spender.name} spends #{@game.format_currency(ability.cost)} and teleports to #{hex.name}"
+          end
+
+          ability.use!
```

Comment:
`process_lay_tile` in `special_track.rb` already calls `ability.use!`. Would it be possible for `lay_tile` to take teleport and cost as parameters, and get rid of the `.abilities(:teleport)` call here?

https://github.com/crericha/18xx/blob/fbc83f1fa66e894c5c883bc643ced201dbaf0e49/lib/engine/step/special_track.rb#L23-L28

Reply:
I didn't add ability.use!. It was already there. (line 94)

There are other games that allow teleport without a tile lay (1849 comes to mind), so I think the teleport ability is still needed?

Reply:
> I didn't add ability.use!. It was already there (line 94). Is it something that should be removed?

Oh sorry, I missed that! Yeah looks like we need to leave it